lib/rule-tester/rule-tester.js

```diff
@@ -621,6 +627,21 @@ class RuleTester {
             ...defaultConfig.slice(1)
         ];
 
+        /**
+         * Runs a hook on the given item when it's assigned to the given property
+         * @param {string|Object} item Item to run the hook on
+         * @param {string} prop The property having the hook assigned to
+         * @throws {Error} If the property is not a function or that function throws an error
+         * @returns {void}
+         * @private
+         */
+        function runHook(item, prop) {
+            if (typeof item === "object" && hasOwnProperty(item, prop)) {
+                assert.ok(typeof item[prop] === "function", `Optional test case property '${prop}' must be a function`);
```

Comment:
Let's use a more specific assert here.
```suggestion
                assert.strictEqual(typeof item[prop], "function", `Optional test case property '${prop}' must be a function`);
```